pyk/src/pyk/cfg_manager.py

```diff
@@ -123,145 +119,6 @@ def _cellVarsToLabels(_kast):
-    return 1
-
-
-def rewriteAndBoolACAnywhereWith(rewrite: KRewrite, term: KInner) -> KInner:
-    config, constraint = split_config_and_constraints(term)
-    constraints = [simplify_bool(ml_pred_to_bool(c, unsafe=True)) for c in flatten_label('#And', constraint)]
-    changed = True
-    while changed:
-        changed = False
-
-        alias_clauses = flatten_label('_andBool_', rewrite.lhs)
-        ac_matches = matchAC(alias_clauses, constraints)
-        if ac_matches is None:
-            continue
-        matches = []
-        all_constraint_ids = []
-        for constraint_ids, subst in ac_matches:
-            if not any(cid in all_constraint_ids for cid in constraint_ids):
-                matches.append(subst)
-                all_constraint_ids.extend(constraint_ids)
-        new_constraints = []
-        for constraint_id, constraint in enumerate(constraints):
-            if constraint_id not in constraint_ids:
-                new_constraints.append(constraint)
-        for subst in matches:
-            new_constraints.append(substitute(rewrite.rhs, subst))
-            changed = True
-        constraints = new_constraints
-
-    return mlAnd([config] + [bool_to_ml_pred(c) for c in constraints])
-
-
-def matchAC(clauses: Sequence[KInner], terms: Sequence[KInner]) -> Optional[List[Tuple[List[int], Subst]]]:
-
-    def _combineClauseMatchPair(
-        match_1: Tuple[List[int], Subst],
-        match_2: Tuple[List[int], Subst],
-    ) -> List[Tuple[List[int], Subst]]:
-        term_ids_1, subst_1 = match_1
-        term_ids_2, subst_2 = match_2
-        if any(cid in term_ids_2 for cid in term_ids_1):
-            return []
-        subst = subst_1.union(subst_2)
-        if subst is None:
-            return []
-        return [(term_ids_1 + term_ids_2, subst)]
-
-    def _combineClauseMatches(
-        matches_1: Sequence[Tuple[List[int], Subst]],
-        matches_2: Sequence[Tuple[List[int], Subst]],
-    ) -> List[Tuple[List[int], Subst]]:
-        clause_matches = []
-        for match_1 in matches_1:
-            for match_2 in matches_2:
-                clause_matches.extend(_combineClauseMatchPair(match_1, match_2))
-        return clause_matches
-
-    missing_clause = False
-    matches: List[List[Tuple[List[int], Subst]]] = [[] for _ in clauses]
-    for i, clause in enumerate(clauses):
-        missing_this_clause = True
-        for j, term in enumerate(terms):
-            clause_match = clause.match(term)
-            if clause_match is not None:
-                missing_this_clause = False
-                matches[i].append(([j], clause_match))
-        missing_clause = missing_clause or missing_this_clause
-        if missing_clause:
-            break
-
-    if missing_clause:
-        return None
-
-    else:
-        full_matches = reduce(_combineClauseMatches, matches)
-        if len(full_matches) == 0:
-            return None
-        return full_matches
-
-
```

Comment:
All this code can now be removed without consequence. If we'd like to bring in AC rewriting at a later point, I think its important to have unit tests for it. I am not comfortable adding it in its current state.